velox/functions/sparksql/SparkPartitionId.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/SparkPartitionId.h"
+
+#include "velox/core/QueryConfig.h"
+#include "velox/expression/VectorFunction.h"
+
+#include "velox/functions/Macros.h"
+#include "velox/functions/Registerer.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+class SparkPartitionId final : public facebook::velox::exec::VectorFunction {
+ public:
+  void apply(
+      const facebook::velox::SelectivityVector& rows,
+      std::vector<facebook::velox::VectorPtr>& /*args*/,
+      const facebook::velox::TypePtr& outputType,
+      facebook::velox::exec::EvalCtx& context,
+      facebook::velox::VectorPtr& result) const override {
+    auto* pool = context.pool();
+    auto partitionId =
+        context.execCtx()->queryCtx()->queryConfig().sparkPartitionId();
+    VELOX_CHECK_GT(partitionId, -1, "invalid spark parititon id");
+    auto localResult = facebook::velox::BaseVector::createConstant(
+        outputType, partitionId, rows.end(), pool);
+    context.moveOrCopyResult(localResult, rows, result);
+  }
+
+  bool isDefaultNullBehavior() const override {
+    return false;
+  }
+
+  static std::vector<std::shared_ptr<facebook::velox::exec::FunctionSignature>>
+  signatures() {
+    return {facebook::velox::exec::FunctionSignatureBuilder()
+                .returnType("integer")
+                .build()};
+  }
+};
+} // namespace
+void registerSparkPartitionId(const std::string& name) {
+  facebook::velox::exec::registerVectorFunction(
+      name,
+      SparkPartitionId::signatures(),
+      std::make_unique<SparkPartitionId>());
```

Comment:
It seems `partitionId` could be a construction parameter of SparkPartitionId function if it's registered as a stateful vector function. Then it is not needed to be accessed through query config in each call of apply.